R/scatter_plot_financial.R

```diff
@@ -0,0 +1,130 @@
+#' Create a scatter plot with financial data
+#'
+#' Generate a scatter plot comparing the emission profile and transition risks
+#' for each bank in the portfolio.
+#'
+#' @param data A data frame like [financial].
+#' @param benchmarks A character vector specifying the benchmarks for which the
+#' emission profiles will be plotted. The user can choose from one to several
+#' benchmark(s) to be plotted.
+#' @param mode A character vector specifying the mode of financial data to plot.
+#' It can be one of "equal_weight", "worst_case" or "best_case". If nothing is
+#' chosen, "equal_weight" is the default case.
+#' @param scenario A character vector specifying the scenario to be plot. It can
+#' either be "IPR" or "WEO".
+#' @param year A numerical specifying the year of the scenario to be plot. It
+#' can be either 2030 or 2050.
+#'
+#' @return A [ggplot] object.
+#' @export
+#'
+#' @examples
+#' scatter_plot_financial(financial, "all", "equal_weight", "IPR", 2030)
```

Comment:
In general it's fine to omit the name of the first argument (easiest to remember) but the second and others should be named explicitely. Pretty sure I read that advice somewhere in https://design.tidyverse.org/ or https://style.tidyverse.org/

```r
scatter_plot_financial(financial, benchmarks = "all", mode = "equal_weight", scenario = "IPR", year = 2030)
```

As I type this I realize that the user should choose only 1 benchmark so the plural "benchmark**s**" seems missleading.

Reply:
@maurolepore Thanks ! Good point, @AnneSchoenauer asked that the user could choose multiple benchmarks for some plots (for example the bar plot.) I will check with her if that is the case for the scatter plot here.